hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleReadyForExecution.java

```diff
@@ -36,17 +46,28 @@
 		this.txnCtx = context;
 	}
 
+	/**
+	 * Given a ScheduleId, check if the underlying transaction
+	 * is already executed/deleted before attempting to execute.
+	 *
+	 * @param id The id of the scheduled transaction.
+	 *
+	 * @return the response code from executing the inner scheduled transaction
```

Comment:
I'd like us to document the expected state and conditions for the param and the return value. I see that `id` is an object, which means it might be null. Is null a valid input? How does it behave if we do pass a null? (Is there a test for this?).

The method documentation suggests that this method checks if the underlying transaction is already executed/deleted. Based on this wording, I would expect the method to return a boolean. But actually, I think this method does more, it just happens to perform this check in addition to other stuff? The method description doesn't seem to align with the method signature.

Reply:
Thanks for reviewing it, I have changed the method documentation, added tests for not null checks. The inputs now have a @NotNull annotation. The return value would always be a proper ResponseEnumCode.